app/model/Suggestion.scala

```diff
@@ -0,0 +1,30 @@
+package model
+
+import play.api.libs.json.{Json, Reads, Writes}
+
+object Suggestion {
+  implicit val writes: Writes[Suggestion] = {
+    case textSuggestion: TextSuggestion =>
+      TextSuggestion.writes.writes(textSuggestion)
+  }
+}
+
+sealed trait Suggestion {
+  val `type`: String
+  val text: String
+}
+
+object TextSuggestion {
+  implicit val reads: Reads[TextSuggestion] = Json.reads[TextSuggestion]
+  implicit val writes = new Writes[TextSuggestion] {
+    def writes(suggestion: TextSuggestion) = Json.obj(
```

Comment:
Why do we need a custom writer? Could we use a [formatter](https://www.playframework.com/documentation/2.7.x/ScalaJsonCombinators#Format) instead?

Reply:
IIRC if we don't create our own, the JSON is missing a `type` parameter. But perhaps there's a more elegant way to do this?

Reply:
To cut down on duplication, for example, we could write ```
def writes(suggestion: TextSuggestion) = Json.obj("type" -> suggestion.`type`) ++ Json.writes[TextSuggestion](suggestion)
```

Reply:
Libraries like `play-json-extensions` provide [formatSealed](https://github.com/xdotai/play-json-extensions#formatsealed-uses-orelse-of-subclass-reads-in-random-order-careful-in-case-of-ambiguities-of-field-class-correspondances) to auto-generate serialisers and deserialisers for sealed traits but I think it's also fine just to manually define.

Probably best to define the Read and Writes on the companion object for the trait though to get exhaustivity checking